connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -185,46 +180,105 @@ public Task newTask(Class<? extends Task> taskClass) {
+            Class<? extends Converter> klass;
+            try {
+                klass = pluginClass(
+                        delegatingLoader,
+                        converterClassOrAlias,
+                        Converter.class
+                );
+            } catch (ClassNotFoundException e) {
+                throw new ConnectException(
+                        "Failed to find any class that implements Converter and which name matches "
+                                + converterClassOrAlias
+                                + ", available converters are: "
+                                + pluginNames(delegatingLoader.converters())
+                );
+            }
+            plugin = newPlugin(klass);
         }
-        return config != null ? newConfiguredPlugin(config, klass) : newPlugin(klass);
+        if (plugin instanceof Configurable) {
+            String configPrefix = classPropertyName + ".";
+            Map<String, Object> converterConfig = config.originalsWithPrefix(configPrefix);
+            ConverterType type = isKeyConverter ? ConverterType.KEY : ConverterType.VALUE;
+            converterConfig.put(ConverterConfig.TYPE_CONFIG, type.getName());
+            ((Configurable) plugin).configure(converterConfig);
+        }
+        return plugin;
     }
 
-    public HeaderConverter newHeaderConverter(String converterClassOrAlias, AbstractConfig config) {
-        Class<? extends HeaderConverter> klass;
-        try {
-            klass = pluginClass(
-                    delegatingLoader,
-                    converterClassOrAlias,
-                    HeaderConverter.class
-            );
-        } catch (ClassNotFoundException e) {
-            throw new ConnectException(
-                    "Failed to find any class that implements HeaderConverter and which name matches "
-                            + converterClassOrAlias
-                            + ", available header converters are: "
-                            + pluginNames(delegatingLoader.converters())
-            );
+    /**
+     * Instantiate and configure a {@link HeaderConverter} identified in the supplied configuration with the given configuration
+     * property name.
+     *
+     * @param config            the configuration containing the {@link HeaderConverter}'s configuration; may not be null
+     * @param classPropertyName the name of the property that contains the name of the {@link HeaderConverter} class;
+     *                          may not be null
+     * @return the instantiated and configured {@link HeaderConverter}; never null
+     * @throws ConnectException if the {@link HeaderConverter} implementation class could not be found
+     */
+    public HeaderConverter newHeaderConverter(AbstractConfig config, String classPropertyName) {
+        // First see if we can instantiate the instance using the current classloader ...
+        HeaderConverter plugin = getInstance(config, classPropertyName, HeaderConverter.class);
```

Comment:
The code isn't huge, so by no means a blocker, but @kkonstantine pointed out that the entire block before configuration (and most of config modulo the conditional) is identical between header and "normal" converters. The main difference in the header blocks are just the class referenced (`Converter.class` vs `HeaderConverter.class`).

Consolidation would be nice if it's easy to do, but at the same time I'd rather get a fix to the immediate problem in, so definitely wouldn't block on saving 10 lines of duplicated code.